lib/os/sem.c

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright (c) 2019 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <misc/sem.h>
+#include <syscall_handler.h>
+
+void sys_sem_init(struct sys_sem *sem)
+{
+	k_futex_init(&sem->futex);
+}
+
+int sys_sem_give(struct sys_sem *sem)
+{
+	int ret = 0;
+
+	atomic_add(&sem->futex.val, 1);
+	ret = k_futex_wake(&sem->futex, false);
+
+	if (ret) {
```

Comment:
What is the purpose of this check?
As far as I can tell, it hides any errors in giving the semaphore.